Packs/AWS-GuardDuty/Integrations/AWSGuardDuty/AWSGuardDuty.py

```diff
@@ -26,297 +21,249 @@ def default(self, obj):  # pylint: disable=E0202
-        if args.get('format') is not None:
-            kwargs.update({'Format': args.get('format')})
-        if args.get('location') is not None:
-            kwargs.update({'Location': args.get('location')})
-        if args.get('name') is not None:
-            kwargs.update({'Name': args.get('name')})
-
-        response = client.create_threat_intel_set(**kwargs)
-
-        data = ({
-            'DetectorId': args.get('detectorId'),
-            'ThreatIntelSetId': response['ThreatIntelSetId']
-        })
-        ec = {
-            "AWS.GuardDuty.Detectors(val.DetectorId === obj.DetectorId).ThreatIntelSet"
-            "(val.ThreatIntelSetId === obj.ThreatIntelSetId)": data}
-        return create_entry('AWS GuardDuty ThreatIntel Set', data, ec)
-
-    except Exception as e:
-        return raise_error(e)
+def create_detector(client: boto3.client, args: dict):
+    kwargs = {'Enable': True if args.get('enable') == 'True' else False}
+    response = client.create_detector(**kwargs)
+    data = ({'DetectorId': response['DetectorId']})
+    readable_output = tableToMarkdown('AWS GuardDuty Detectors', data) if data else 'No result were found'
+    return CommandResults(readable_output=readable_output,
+                          outputs=data,
+                          outputs_prefix='AWS.GuardDuty.Detectors',
+                          outputs_key_field='DetectorId')
 
 
-def delete_threat_intel_set(client, args):
-    try:
-        response = client.delete_threat_intel_set(
-            DetectorId=args.get('detectorId'),
-            ThreatIntelSetId=args.get('threatIntelSetId')
-        )
-        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
-            return 'The ThreatIntel Set {0} has been deleted from Detector {1}'.format(args.get('ipSetId'),
-                                                                                       args.get('detectorId'))
-
-    except Exception as e:
-        return raise_error(e)
-
-
-def get_threat_intel_set(client, args):
-    try:
-        response = client.get_threat_intel_set(
-            DetectorId=args.get('detectorId'),
-            ThreatIntelSetId=args.get('threatIntelSetId')
-        )
-        data = ({
-            'DetectorId': args.get('detectorId'),
-            'ThreatIntelSetId': args.get('threatIntelSetId'),
-            'Format': response['Format'],
-            'Location': response['Location'],
-            'Name': response['Name'],
-            'Status': response['Status']
-        })
-        ec = {
-            "AWS.GuardDuty.Detectors(val.DetectorId === obj.DetectorId).ThreatIntelSet"
-            "(val.ThreatIntelSetId === obj.ThreatIntelSetId)": data}
-        return create_entry('AWS GuardDuty ThreatIntel Set', data, ec)
-
-    except Exception as e:
-        return raise_error(e)
+def delete_detector(client: boto3.client, args: dict):
+    response = client.delete_detector(DetectorId=args.get('detectorId'))
+    if response == dict():
```

Comment:
Why the change from this? `if response['ResponseMetadata']['HTTPStatusCode'] == 200:`

Reply:
When changing to python3 the boto3 client library changed to its py3 version
The new SDK responses do not include the 'ResonseMetadata' section anymore.
You can find documentation for the new responses structure in the following:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/guardduty.html#